In [1]:
import numpy as np

In [2]:
M = np.array([[1,2], [3,4]])

In [3]:
L = [[1,2],[3,4]]

In [4]:
L[0]

[1, 2]

In [5]:
L[0][0]

1

In [6]:
M[0][0]

1

In [7]:
M[0,0]

1

In [8]:
M2 = np.matrix([[1,2], [3,4]])

In [9]:
M2


matrix([[1, 2],
        [3, 4]])

In [11]:
A = np.array(M2)

In [12]:
A

array([[1, 2],
       [3, 4]])

In [13]:
A.T

array([[1, 3],
       [2, 4]])

In [15]:
np.array([1,2,3])

array([1, 2, 3])

In [17]:
Z = np.zeros(10)

In [18]:
Z

array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0.])

In [19]:
Z = np.zeros((10, 10))

In [21]:
Z

array([[0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.]])

In [22]:
R = np.random.random((10,10))

In [23]:
R

array([[0.3028515 , 0.45878876, 0.26221228, 0.90588055, 0.37445284,
        0.65273674, 0.50599484, 0.17440798, 0.30448442, 0.00947806],
       [0.01575736, 0.88814625, 0.70807683, 0.99356605, 0.37417594,
        0.40824687, 0.78562839, 0.88660853, 0.2849974 , 0.33179821],
       [0.54330964, 0.11237675, 0.43327844, 0.98445211, 0.5399331 ,
        0.74495256, 0.56959472, 0.68470984, 0.40811724, 0.62154743],
       [0.11080659, 0.49203666, 0.58376844, 0.70526455, 0.59359115,
        0.12609059, 0.54093053, 0.44641163, 0.22468206, 0.88031496],
       [0.76367261, 0.62206234, 0.13677878, 0.07497822, 0.28754404,
        0.92321628, 0.13226124, 0.48708865, 0.15264615, 0.8157199 ],
       [0.54526251, 0.50902895, 0.50050557, 0.34404594, 0.48279685,
        0.40078033, 0.70714124, 0.62328868, 0.94437359, 0.90922723],
       [0.03955534, 0.17422163, 0.53402646, 0.41383755, 0.42121821,
        0.348517  , 0.36105215, 0.11907427, 0.18067809, 0.10785278],
       [0.97011421, 0.5497919 , 0.9812517

In [24]:
G = np.random.randn(10,10)

In [25]:
G

array([[-0.15674385, -0.68507479, -0.52374196, -1.00505187, -1.10891805,
         1.27514623, -1.65116737,  0.68173055,  0.57087671, -0.03698247],
       [-0.36725261, -1.19561363, -1.30670275,  1.16675661,  0.22917048,
        -0.54116151,  0.24068607,  0.21791734,  0.54521678,  0.90807232],
       [ 0.42905061, -0.06363018, -1.43122313, -0.02710009,  0.64646982,
         0.23594111,  0.13296432, -0.28721582, -0.3458594 ,  1.16302384],
       [ 0.2489232 , -0.03920387,  0.12308591, -0.21592247, -0.30261896,
         1.1093762 ,  0.26511937,  0.11398637,  1.68027674, -0.52371912],
       [ 1.98056064, -0.35777618,  0.42644307,  0.37838021, -0.45527457,
        -0.49813577, -0.38937133,  0.6732684 ,  1.00203298,  0.14786914],
       [-1.11274666, -1.31320159,  0.7185951 ,  2.0454256 ,  1.22951562,
         1.17904756,  0.90509421, -1.06226059, -0.04086287,  0.55370033],
       [ 0.70122436,  0.71639924,  0.93603808,  2.74241062,  0.21745398,
         0.23045127, -1.499646  ,  1.42642966

In [26]:
G.mean()

0.10753985967392272

In [27]:
G.var()

0.7772632421805669

In [28]:
A = np.array([[1,2], [3,4]])

In [29]:
A

array([[1, 2],
       [3, 4]])

In [30]:
ainv = np.linalg.inv(A)

In [31]:
ainv

array([[-2. ,  1. ],
       [ 1.5, -0.5]])

In [32]:
np.linalg.det(A)

-2.0000000000000004

In [33]:
np.diag(A)

array([1, 4])

In [34]:
np.diag([1,2])

array([[1, 0],
       [0, 2]])

In [35]:
a = np.array([1,2])

In [36]:
b = np.array([3,4])

In [37]:
np.outer(a,b)

array([[3, 4],
       [6, 8]])

In [38]:
np.inner(a,b)

11

In [39]:
a.dot(b)

11

In [40]:
np.diag(A).sum()

5

In [41]:
np.trace(A)

5

In [42]:
x = np.random.randn(100,3)

In [43]:
cov = np.cov(x)

In [44]:
cov.shape

(100, 100)

In [45]:
cov = np.cov(x.T)

In [46]:
cov

array([[ 1.23760448, -0.05259634,  0.17112741],
       [-0.05259634,  1.06508062, -0.06560082],
       [ 0.17112741, -0.06560082,  0.87207636]])

In [47]:
np.linalg.eigh(cov)

(array([0.79770145, 1.05113845, 1.32592155]),
 array([[ 0.34117778,  0.31248117, -0.88654004],
        [-0.16014162,  0.94866366,  0.27274884],
        [-0.92625719, -0.04891611, -0.3737042 ]]))

In [49]:
np.linalg.eig(cov)

(array([1.32592155, 0.79770145, 1.05113845]),
 array([[-0.88654004,  0.34117778,  0.31248117],
        [ 0.27274884, -0.16014162,  0.94866366],
        [-0.3737042 , -0.92625719, -0.04891611]]))

In [50]:
a

array([1, 2])

In [51]:
A

array([[1, 2],
       [3, 4]])

In [52]:
b

array([3, 4])

In [53]:
b = np.array([1,2])

In [54]:
x = np.linalg.solve(A,b)

In [55]:
x

array([2.46716228e-17, 5.00000000e-01])

In [56]:
A

array([[1, 2],
       [3, 4]])

In [57]:
b

array([1, 2])

In [58]:
x = np.linalg.solve(A,b)

In [59]:
x

array([2.46716228e-17, 5.00000000e-01])

In [60]:
x = np.linalg.inv(A).dot(b)

In [61]:
x

array([0. , 0.5])

In [63]:
np.linalg.solve(A, b)

array([2.46716228e-17, 5.00000000e-01])

The admission fee at a small fiar is $1.50 for children and $4.00 for adults. On a Certain day, 2200 people enter the fair and $5050 is collected. How many children and how many adults attended?

Let

X1 = number of children, X2 number of adults
X1 + x2 = 2200
1.5X1 + 4X2 = 5050

In [66]:
A = np.array([[1,1], [1.5, 4]])

In [67]:
b = np.array([2200, 5050])

In [69]:
np.linalg.solve(A,b)

array([1500.,  700.])